ai/main.py

```
@@ -4,7 +4,21 @@
+@app.post("/v1/book/{book_id}")
+def synopsys(book_id: int, authorization: str = Header(None)):
+    try:
+        content = get_book(book_id, authorization)
+        if content:
+            content = content
+            result = ai.generate_synopsys(book_id, authorization,content)
+            return {"status": "success", "result":result}
+        else:
+            raise HTTPException(status_code=404,detail="데이터가 없습니다.")
+    except Exception as e:
+        raise HTTPException(status_code=500,detail=f"DB 연결 실패: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)